onadata/apps/logger/models/instance.py

```diff
@@ -389,6 +386,7 @@ def get_full_dict(self, load_existing=True):
                 self.date_created = submission_time()
 
             doc[SUBMISSION_TIME] = self.date_created.strftime(MONGO_STRFTIME)
+            doc[DATE_MODIFIED] = self.date_modified.strftime(MONGO_STRFTIME)
```

Comment:
We do have this check https://github.com/onaio/onadata/pull/2009/files#diff-98ae21dd41854ad6194bdbca52b97c4d2b0231f6d06b3ee5ee73c4c88a135e27R385-R386 for `date_created` ensuring it is not null, should we have the same for `date_modified`?